src/main/scala/mesosphere/marathon/state/Volume.scala

```diff
@@ -33,33 +36,21 @@ object Volume {
         )
     }
 
-  def apply(proto: Protos.Volume): Volume = {
-    val persistent: Option[PersistentVolumeInfo] =
-      if (proto.hasPersistent) Some(PersistentVolumeInfo(proto.getPersistent.getSize)) else None
-
-    persistent match {
-      case Some(persistentVolumeInfo) =>
-        PersistentVolume(
-          containerPath = proto.getContainerPath,
-          persistent = persistentVolumeInfo,
-          mode = proto.getMode
-        )
-      case None =>
-        DockerVolume(
-          containerPath = proto.getContainerPath,
-          hostPath = proto.getHostPath,
-          mode = proto.getMode
-        )
-    }
+  def fromProto(proto: Protos.Volume): Volume = {
```

Comment:
would rather move this into the volume module and let provider implementations deal with this